tests/src/LinkParser.cpp

```diff
@@ -89,14 +89,21 @@ TEST(LinkParser, parseDomainLinks)
         {"", "https.cat"},
         {"", "httpsd.cat"},
         {"", "http.cat", "/200"},
-        {"", "http.cat", "/200("},
-        {"", "a.com", "?("},
-        {"", "a.com", "#("},
+        {"", "http.cat", "/200()"},
+        {"", "a.com", "?()"},
+        {"", "a.com", "#()"},
```

Comment:
We can't have test cases that don't have balanced parentheses, as all possible suffixes are checked and having a suffix that balances the parentheses would result in an error (e.g. `a.com#(` with `).` would result in `a.com#()` being the link and `.` being the suffix).